### PR TITLE
Initial project settings with copy @dooboolab/react-native-iap

### PR DESCRIPTION
**Pull request by mym0404**
* Copy initial project settings from [@dooboolab/react-native-iap](https://github.com/dooboolab/react-native-iap)
  - **Copy** `eslint`, `gitignore`, `ncurc`, `license`, `tsconfig` etc...
  - **Update** version of dependencies to latest with ncu

* Create **useApi** directory for `useApi` hooks

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -0,0 +1,10 @@
+IapExample/
+.idea/
+gen/
+node_modules/
+android/react-native-iap.iml
+android/android.iml
+tsconfig.tsbuildinfo
+index.ts
+tsconfig.json
+greenkeeper.json
```
Comment (hyochan) [inline .npmignore]: IapExample/
gen/
android/react-native-iap.iml
android/android.iml
greenkeeper.json

It looks redundant here. We are not developing any natives and not integrating greenkeeper!

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2017-2018 dooboolab
```
Comment (hyochan) [inline LICENSE]: Copyright 2020

File: package.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    "url": "git+https://github.com/dooboolab/dooboo-hooks"
+  },
+  "keywords": [
+    "react-native"
+  ],
+  "author": "dooboolab",
+  "license": "MIT",
+  "peerDependencies": {
+    "react": "*",
+    "react-native": ">=0.54"
```
Comment (hyochan) [inline package.json]: Since we are supporting hook, we need `"react-native": ">=0.59"`

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,60 @@
+
+# OSX
+#
+.DS_Store
+
+# node.js
+#
+node_modules/
+package-lock.json
+IapExample/package-lock.json
```
Comment (hyochan) [inline .gitignore]: Please remove this line.

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+android/.project
+
+# BUCK
+buck-out/
+\.buckd/
+*.keystore
+
+# CocoaPods
+IapExample/ios/Pods
+react-native-iap-*.tgz
```
Comment (hyochan) [inline .gitignore]: Redundant~! I think you can remove all of these lines.

File: .gitignore (hunk 4 above)
Comment (mym0404) [reply .gitignore]: @hyochan  why do we remove `node_modules/` in `.gitignore`? 
I have thought about the existence of `node_modules/` in `.gitignore` prevents repository containing so many modules code.

Comment (hyochan) [reply .gitignore]: Oh I've missed this!! Thanks. Please keep it!

File: jest.config.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  preset: 'ts-jest',
+  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
+  testPathIgnorePatterns: ['node_modules', 'lib'],
```
Comment (mym0404) [inline jest.config.js]: I have to change typo lib to dist

**Pull request merged**